### ISSUE TITLE
lichess support

### ISSUE THREAD

**Issue by JoelMon** [pychess/pychess#1332]
Would be wonderful if apart from playing on FICS that we can also play on lichess.org.

The HTTP API is documented on the project page. https://github.com/ornicar/lila. Ornicar, the developer of lichess, is also happy to provide help with the API.

There's more documentation [here](https://github.com/ornicar/lila/tree/master/doc/mobile). It might not be up to date though. Another source of information might be the codebase for the lichess mobile app, [lichobile](https://github.com/veloce/lichobile).

**Comment by gbtami** [comment]
These API-s are very limited (only get players/games). If they will provide more API to _play_ games we can add support for it.
On the other side lichess.org has a wonderful web GUI. May I ask you why do you want to use anything else?

**Comment by JoelMon** [comment]
I would use it to practice something against the computer and then login to lichess through PyChess and play. It's not a priority kind of thing but it's nice. Also there's a nice community coming around lichess, maybe supporting lichess might bring more people to PyChess. My last reason is that I know a little Python and would love to see how it's implemented. :smile:  lol

The API to play games on lichess is there, I think it just hasn't been documented formally. lichmobile makes use of it and it's 100% playable. The API to play a game is here: https://github.com/ornicar/lila/blob/master/doc/mobile/play.md

Here's another request for something similare: https://github.com/veloce/lichobile/issues/244

**Comment by gbtami** [comment]
Cool! I totally missed that part of ornicar/lila doc. Thx!
All in all, as Thomas mentioned in https://groups.google.com/forum/#!searchin/pychess-people/lichess/pychess-people/Tqb0If36qYo/2hD12Ib0IgAJ "we're interested" :)

**Comment by ghost** [comment]
"Diagram of lichess architecture"
https://twitter.com/lichessorg/status/685024931583672320/photo/1

**Comment by JoelMon** [comment]
@gbtami:  Wonderful, when work begins on it can you let me know? I would like to follow along to see how you guys go about doing it.

Thanks

**Comment by Unihedro** [comment]
about the limited API, endpoints aren't made on lichess unless someone suggests how they can be useful and will use them. This is probably to reduce the burden of having to maintain a full API in parallel to the constantly improved codebase. A recent (?) discussion on this topic may interest you: ornicar/lila#1253

**Comment by gbtami** [comment]
@JoelMon sure:)

@Unihedro I absolutely agree with your decision. And  thx for the link!

**Comment by gbtami** [comment]
https://lichess.org/api

**Comment by gbtami** [comment]
Seems this will not happen :(
```
<devurand1m> so there's no objection to developing a cross-platform desktop application due to some (possibly security related) lichess policy?
<ornicar> devurand1m: You *must* use the bot API to build a new lichess client.
<ornicar> Anything else will be detected as cheat
<devurand1m> why? seeing that i could modigy the mobile application code (which was the initial idea, yet seemed unclean)
* pionen (~porfirio@fedora/porfiriopaiz) has joined
<devurand1m> too much coke in my keyboard.. damn
<ornicar> because of cheats. A part of detection resides in the clients
<ornicar> Not all of it, but some of it, and we won't give up on it
<devurand1m> again: how does "acid ape chess" then?
<ornicar> it's an exception
```

**Comment by pnmadelaine** [comment]
It seems like third party clients are now allowed by the lichess api : https://lichess.org/api#tag/Board
